source/broadband/webpa_rbus.c

```diff
@@ -0,0 +1,126 @@
+#include <stdbool.h>
+#include <string.h>
+
+#include <stdlib.h>
+#include <wdmp-c.h>
+#include <cimplog.h>
+#include "webpa_rbus.h"
+
+static rbusHandle_t rbus_handle;
+static bool isRbus = false;
+
+bool isRbusEnabled()
+{
+        if(RBUS_ENABLED == rbus_checkStatus())
+        {
+                isRbus = true;
+        }
+        else
+        {
+                isRbus = false;
+        }
+        WalInfo("Webpa RBUS mode active status = %s\n", isRbus ? "true":"false");
+        return isRbus;
+}
+
+bool isRbusInitialized()
+{
+    return rbus_handle != NULL ? true : false;
+}
+
+WDMP_STATUS webpaRbusInit(const char *pComponentName)
+{
+        int ret = RBUS_ERROR_SUCCESS;
+
+        WalInfo("rbus_open for component %s\n", pComponentName);
+        ret = rbus_open(&rbus_handle, pComponentName);
+        if(ret != RBUS_ERROR_SUCCESS)
+        {
+                WalError("webpaRbusInit failed with error code %d\n", ret);
+                return WDMP_FAILURE;
+        }
+        WalInfo("webpaRbusInit is success. ret is %d\n", ret);
+        return WDMP_SUCCESS;
+}
+
+static void webpaRbus_Uninit()
+{
+    rbus_close(rbus_handle);
+}
+
+rbusError_t setTraceContext(char* traceContext[])
+{
+        rbusError_t ret = RBUS_ERROR_BUS_ERROR;
+        if(isRbusInitialized)
+        {
+                if(traceContext[0] != NULL && traceContext[1] != NULL) {
+                       if(strlen(traceContext[0]) > 0 && strlen(traceContext[1]) > 0) {
+			    WalInfo("Invoked setTraceContext function with value traceParent - %s, traceState - %s\n", traceContext[0], traceContext[1]);    
+                            ret = rbusHandle_SetTraceContextFromString(rbus_handle, traceContext[0], traceContext[1]);
+                            if(ret == RBUS_ERROR_SUCCESS) {
+                                  WalInfo("SetTraceContext request success\n");
+                            }
+                             else {
+                                   WalError("SetTraceContext request failed with error code - %d\n", ret);
+                             }
+                        }
+                        else {
+                              WalError("Header is empty\n");
+                        }
+                  }
+                  else {
+                        WalError("Header is NULL\n");
+                  }
+        }
+        else {
+                WalError("Rbus not initialzed in setTraceContext function\n");
+        }	
+        return ret;
+}
+
+rbusError_t getTraceContext(char* traceContext[])
+{
+        rbusError_t ret = RBUS_ERROR_BUS_ERROR;
+        char traceParent[512] = {'\0'};
```

Comment:
Please check the length of 512 is same as used in rbus. Need to confirm on the length

Reply:
yes, in rbus samples they used 512 only the size for traceParent & traceState